src/PetriEngine/Structures/PotencyQueue.cpp

```diff
@@ -141,5 +141,83 @@ namespace PetriEngine {
             _currentParentDist = e.weight;
             return e.item;
         }
+
+        RandomWalkPotencyQueue::RandomWalkPotencyQueue(size_t seed) : PotencyQueue(seed), _seed(seed) {
+            srand(_seed);
+        }
+
+        RandomWalkPotencyQueue::~RandomWalkPotencyQueue() {}
+
+        void
+        RandomWalkPotencyQueue::push(size_t id, PQL::DistanceContext *context, const PQL::Condition *query, uint32_t t) {

```

Comment:
Same function as RandomPotencyQueue::push

Reply:
IF the functionality is the same as the RandomPotencyQueue, then just use that one?

Reply:
I don't know how to use the RandomPotencyQueue push if it's not inherited from it